system/ee/ExpressionEngine/Addons/member/mod.member.php

```diff
@@ -1291,6 +1291,43 @@ public function registration_form()
         return $MR->registration_form();
     }
 
+    /**
+     * ReCaptcha Check
+     * Checks to see if the ReCaptcha Score was valid and strong enough.
+     */
+    public function recaptcha_check()
+    {
+        $recaptcha_url = 'https://www.google.com/recaptcha/api/siteverify';
+        $recaptcha_threshold = ee()->config->item('recaptcha_score_threshhold');
+
+        // Make the POST request
+        $data = [
+            'secret' => ee()->config->item('recaptcha_site_secret'),
+            'response' => ee()->input->get_post('rec'),
+        ];
+
+        $curl = curl_init();
+        curl_setopt($curl, CURLOPT_POST, true);
+        curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, false);
+        curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($curl, CURLOPT_URL, $recaptcha_url);
+        curl_setopt($curl, CURLOPT_POSTFIELDS, http_build_query($data));
+        $response = curl_exec($curl);
+        $result = json_decode($response, true);
+
+        // Mostly random string
+        $string = bin2hex(openssl_random_pseudo_bytes(10));
+
+        $captcha = ee('Model')->make('Captcha');
+        $captcha->date = ee()->localize->now;
+        $captcha->ip_address = ee()->input->ip_address();
+        $captcha->word = $string;
+        $captcha->save();
+
+        if($result['success'] !== true || $result['score'] < $recaptcha_threshold) { $string = "failed"; }
```

Comment:
I think the formatting of these lines could be cleaned up a bit. Putting an entire conditional on a single line isn't PSR

Reply:
Updated